pkg/client/typed/discovery/discovery_client.go

```diff
@@ -209,75 +211,93 @@ func IsGroupDiscoveryFailedError(err error) bool {
 
 // serverPreferredResources returns the supported resources with the version preferred by the
 // server. If namespaced is true, only namespaced resources will be returned.
-func (d *DiscoveryClient) serverPreferredResources(namespaced bool) ([]schema.GroupVersionResource, error) {
+func (d *DiscoveryClient) serverPreferredResources(namespaced bool) ([]*unversioned.APIResourceList, error) {
 	// retry in case the groups supported by the server change after ServerGroup() returns.
```

Comment:
... and reverted. The tests check against 500 errors. This is clearly out of scope of adding verbs. I have added a TODO.